[DNM] feat(api): add commands plugin.

wdstar opened the pull request:
## Context

Screwdriver Commands feature.

## Objective (phase 1)

Add the first commands (API) plugin.

- POST `commands/`: create a command meta data.
- GET `commands/namespace/name/versionOrTag`: retrieve a command meta data.
- without authorization.

## Note

- blocked by https://github.com/screwdriver-cd/models/pull/210

## To do before merged

- [ ] bump up the dependent `screwdriver-models` version.

## References

- [Design](https://github.com/screwdriver-cd/screwdriver/blob/master/design/commands.md)

tkyi committed:
feat(959): Adding guid to JWT and changing timeout to 2h

tkyi committed:
fix: Change ttl to 2h

tkyi committed:
fix: Use uuid instead of aguid

d2lam committed:
Merge branch 'master' into jwt

d2lam committed:
Merge pull request #825 from screwdriver-cd/jwt

fix: Adding guid to JWT and changing timeout to 2h

wdstar committed:
feat(api): add commands plugin.

kumada626 commented on plugins/commands/create.js:
```diff
@@ -0,0 +1,74 @@
+const boom = require('boom');
+// const schema = require('screwdriver-data-schema');
+const validator = require('screwdriver-command-validator');
+// const commandSchema = schema.api.commandValidator;
```
他にこのファイル内でschemaを使用している部分も無さそうなのでL4と合わせて1行にしてしまっても良さそうです

wdstar replied:
確かに。
`commandSchema` 変数を削除して、71行目の参照部分を `schema.api.commandValidator.input` にしたいと思います。
この辺のコードをアンコメントするタイミングで修正します。

jithine committed:
feat(827): add config for tolerations (#828)

* feat(827): add config for tolerations

* fix: correct format for array values

wdstar committed:
fix: rename `command`  key to `name`. (#829)

`command` key of a `command` object/configuration seems to provoke bugs easily.
We should rename  `command`  key to `name`.

jithine committed:
fix(827) update config name (#830)

* fix(827) update config name

* expand comment

* expand comment

jithine committed:
feat: Move to new workflow (#831)

* feat: Move to new workflow

Also add dependency on executor-k8s-vm

* Update screwdriver.yaml

jithine committed:
fix: add config for sequalize connection pool (#833)

* fix: add config for sequalize connection pool

* fix: update doc

jithine committed:
fix: correct config location for nodeSelectors (#832)

d2lam committed:
fix: change uuid to random instead of timestamp (#834)

tkyi committed:
fix: Only return pipelines that match the cluster's scmContexts

petey committed:
Merge pull request #835 from screwdriver-cd/scmContext

fix: Only return pipelines that match the cluster's scmContexts

sakka2 commented on plugins/commands/create.js:
```diff
@@ -0,0 +1,74 @@
+    config: {
+        description: 'Create a new command',
+        notes: 'Create a specific command',
+        tags: ['api', 'commands'],
```
authの設定とかは不要でしょうか？
https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/jobs/update.js#L15

wdstar replied:
はい、最終的には必要ですがPOに確認して今フェイズではスコープ外となりました。

sakka2 replied:
なるほど、了解ですー

d2lam committed:
fix: bumping timeout for sd-step

d2lam committed:
Merge pull request #837 from screwdriver-cd/d2lam-patch-1

fix: bumping timeout for sd-step

wdstar committed:
Merge branch 'master' into commands

wdstar committed:
feat(api): add helpful comments for the next implementation of the commands plugin.
